### PR TITLE
Correctly enforce certificate constraints on all identities

### PR DESCRIPTION
**Pull request by rnicoll**
Enforce that the identity service must always have a root CA specified, which all identities have
certificates signed by (or intermediaries of). Also adds a certificate store to the identity service
for help building/verifying certificate paths.

### CODE REVIEW

File: core/src/test/kotlin/net/corda/core/serialization/KryoTests.kt [hunk 1]
```diff
@@ -152,10 +151,11 @@ class KryoTests {
 
     @Test
     fun `serialize - deserialize X509CertPath`() {
+        val certFactory = CertificateFactory.getInstance("X509")
         val rootCAKey = Crypto.generateKeyPair(X509Utilities.DEFAULT_TLS_SIGNATURE_SCHEME)
         val rootCACert = X509Utilities.createSelfSignedCACertificate(ALICE.name, rootCAKey)
         val certificate = X509Utilities.createCertificate(CertificateType.TLS, rootCACert, rootCAKey, BOB.name, BOB_PUBKEY)
-        val expected = X509Utilities.createCertificatePath(rootCACert, certificate, revocationEnabled = false)
+        val expected = certFactory.generateCertPath(listOf(rootCACert.cert, certificate.cert))
```
Comment (patrickkuo) [inline core/src/test/kotlin/net/corda/core/serialization/KryoTests.kt]: root cert should come last

File: core/src/main/kotlin/net/corda/core/utilities/TestConstants.kt [hunk 2]
```diff
@@ -70,7 +74,8 @@ val DUMMY_CA: CertificateAndKeyPair by lazy {
  * Build a test party with a nonsense certificate authority for testing purposes.
  */
 fun getTestPartyAndCertificate(name: X500Name, publicKey: PublicKey, ca: CertificateAndKeyPair = DUMMY_CA): PartyAndCertificate {
-    val cert = X509Utilities.createCertificate(CertificateType.IDENTITY, ca.certificate, ca.keyPair, name, publicKey)
-    val certPath = X509Utilities.createCertificatePath(ca.certificate, cert, revocationEnabled = false)
-    return PartyAndCertificate(name, publicKey, cert, certPath)
+    val certFactory = CertificateFactory.getInstance("X509")
+    val certHolder = X509Utilities.createCertificate(CertificateType.IDENTITY, ca.certificate, ca.keyPair, name, publicKey)
+    val certPath = certFactory.generateCertPath(listOf(certHolder.cert, ca.certificate.cert))
```
Comment (patrickkuo) [inline core/src/main/kotlin/net/corda/core/utilities/TestConstants.kt]: the path should include root cert?

**Comment by rnicoll** [comment]
CI passed, comments addressed, rebase done. Can we get this merged once M13 is cut?

**Comment by mikehearn** [comment]
If you add new stuff to IdentityService (public API) can you please ensure there are javadocs for them?

**Pull request merged**